test/runner.js

```diff
@@ -327,6 +338,19 @@ describe('Runner', function(){
       fail.should.throw('allow unhandled errors');
       done();
     });
+
+    it('should allow unhandled errors in hooks to propagate through', function(done) {
```

Comment:
this doesn't appear to be async, so you don't need `done`.

Ideally, I'd like to see both sync and async tests against this behavior.  Can you please add an async version?